docs/design/storage.md

```diff
@@ -0,0 +1,99 @@
+  // +optional
+  Dependences []string `json:"dependences,omitempty"`
+  // Storage represents the type of storage for operation results.
+  // Operation results will not be stored if nil.
+  // +optional
+  Storage *Storage `json:"storage,omitempty"`
+}
+
+// Storage represents the type of storage for operation results.
+type Storage struct {
+  // Type specify which storage to save object.
+  // Only support minio now.
+  Type string `json:"type"`
+  // Key specify the scrap key to get file.
+  Key string `json:"key"`
+}
+```
+
+The storage is designed for users to define how should KubeDiag store their data. It contains the following fields:
+
+* `spec.storage.type`: VolumeType should be declared as `minio` in operation.
+* `spec.storage.key`: Key indicates the file path key that needs to be stored.
+
+## Example
+
+Here is example to store the file generated by go profiler operation in minio.
+
+```yaml
+apiVersion: diagnosis.kubediag.org/v1
+kind: Operation
+metadata:
+  name: go-profiler
+spec:
+  processor:
+    path: /processor/goProfiler
+    scheme: http
+    timeoutSeconds: 60
+  storage: 
+    type: minio
+    key: diagnoser.runtime.go_profiler.result.path
+```
+
+After diagnosis，the result generated by go profiler processor contains keys `diagnoser.runtime.go_profiler.result.path`. The value of key `diagnoser.runtime.go_profiler.result.path` corresponds to the generated file path, which will be obtained by agent and stored as file in the form of a key-value pair, with `<bucket_name>/<diagnosis_namespace>.<diagnosis_name>.<node_name>.<pod_namespace>.<pod_name>.<container_name>.<timestamp>.<file_name>` as key and file as value. Undefined values in key are represented by `0`.
+
+```yaml
+apiVersion: diagnosis.kubediag.org/v1
+kind: Diagnosis
+metadata:
+  annotations:
+    storage.operation.go_profiler.stored.file: go-profiler/default.go-profiler.my-node.0.0.0.20210728084437.default.go-profiler.heap.prof
+  # ......
+  name: go-profiler
+  namespace: default
+spec:
+  nodeName: my-node
+  operationSet: go-profiler
+  parameters:
+    param.diagnoser.runtime.go_profiler.expiration_seconds: "7200"
+    param.diagnoser.runtime.go_profiler.source: https://10.0.2.15:6443
+    param.diagnoser.runtime.go_profiler.tls.secret_reference.name: apiserver-profiler-sa-token-wqnr2
+    param.diagnoser.runtime.go_profiler.tls.secret_reference.namespace: kubediag
+    param.diagnoser.runtime.go_profiler.type: Heap
+status:
+  phase: Succeeded
+  operationResults:
+    diagnoser.runtime.go_profiler.result.path: /var/lib/kubediag/profilers/go/pprof/20210728084437/default.go-profiler.heap.prof
+  # ......
+```
+
+After successful storage, the annotation of this diagnosis will record the stored file in minio is `go-profiler/default.go-profiler.my-node.0.0.0.20210728084437.default.go-profiler.heap.prof`. If the stored procedure fails, the value corresponding to this annotation is `<none>`.
```

Comment:
If the annotation is not a conventional API, it should not be illustrated in the document for a high level design of storage.